File: recursion/cpp/09_reverse_string.cpp
#include <bits/stdc++.h>
using namespace std;

void print_reverse(string &st, int n)
{
    if (n == 0)
    {
        return;
    }
    cout << (st[n - 1]);
    print_reverse(st, n - 1);
}

string reverse(string st, int n)
{
    if (n == 0)
    {
        return "";
    }
    return st[n - 1] + reverse(st, n - 1);
}

int main()
{
    string st;
    cin >> st;

    print_reverse(st, st.length());
    cout << "\n";

    cout << reverse(st, st.length());
    cout << "\n";
}

File: days/recursion/py/09_reverse_string.py
def print_reverse(st: str, n: int) -> None:
    if n == 0:
        return
    print(st[n-1], end="")
    print_reverse(st, n-1)


def reverse(st: str, n: int) -> str:
    if n == 0:
        return ""
    return st[n-1] + reverse(st, n-1)


st = input()
print_reverse(st, len(st))
print(reverse(st, len(st)))
